source/Ship.cpp

```diff
@@ -1079,6 +1079,20 @@ int64_t Ship::ChassisCost() const
 
 
 
+// Get the full cost of this ship on the planet, depending on the outfit prices.
+int64_t Ship::LocalCost(const Planet *planet) const
+{
+	int64_t localCost = ChassisCost();
+	for(const auto &it : this->Outfits())
+	{
+		int64_t outfitCost = planet ? planet->Outfitter().GetCost(it.first) : 0.;
+		localCost += (outfitCost ? outfitCost : it.first->Cost()) * it.second;
```

Comment:
Why do you have two conditionals? `planet ? planet->Outfitter().GetCost(it.first) : it.first->Cost()`